tv_apps/browser/js/sync/db.js

```diff
@@ -1190,14 +1190,34 @@ SyncBrowserDB.db = {
     var records = [];
     var db = this._db;
 
-    var transaction = db.transaction(DBOS_BOOKMARKS);
-    var objectStore = transaction.objectStore(DBOS_BOOKMARKS).index('parentid');
+    function makeBookmarkProcessor(data) {
+      if(data.type !== 'bookmark') {
+        records.push(data);
+      } else {
+        iconsStore.get(data.bmkUri).onsuccess = function(e) {
+          var icons = e.target.result;
+          if(icons){
+            if(icons.iconUri){
+              data.iconUri = icons.iconUri;
+            } else {
+              data.iconUri = Awesomescreen.DEFAULT_FAVICON;
+            }
+          }
+          records.push(data);
+        };
+      }
+    }
+
+    var transaction = db.transaction([DBOS_BOOKMARKS, DBOS_ICONS]);
+    var bookmarksStore =
+      transaction.objectStore(DBOS_BOOKMARKS).index('parentid');
+    var iconsStore = transaction.objectStore(DBOS_ICONS);
 
-    objectStore.openCursor(parentId, 'prev').onsuccess =
+    bookmarksStore.openCursor(parentId, 'prev').onsuccess =
       function onSuccess(e) {
       var cursor = e.target.result;
       if (cursor) {
-        records.push(cursor.value);
+        makeBookmarkProcessor(cursor.value);
```

Comment:
makeBookmarkProcessor should return Promise to handle the following tasks - records.push(cursor.value); cursor.continue();